src/Symfony/Component/Serializer/Tests/Fixtures/metadata.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+// This file has been auto-generated by the Symfony Serializer Component.
+
+return array(
+ 'Symfony\Component\Serializer\Tests\Fixtures\Dummy' => \Symfony\Component\VarExporter\Internal\Hydrator::hydrate(
+    $o = [
+        clone (($p = &\Symfony\Component\VarExporter\Internal\Registry::$prototypes)['Symfony\\Component\\Serializer\\Mapping\\ClassMetadata'] ?? \Symfony\Component\VarExporter\Internal\Registry::p('Symfony\\Component\\Serializer\\Mapping\\ClassMetadata')),
+        clone ($p['Symfony\\Component\\Serializer\\Mapping\\AttributeMetadata'] ?? \Symfony\Component\VarExporter\Internal\Registry::p('Symfony\\Component\\Serializer\\Mapping\\AttributeMetadata')),
+        clone $p['Symfony\\Component\\Serializer\\Mapping\\AttributeMetadata'],
+        clone $p['Symfony\\Component\\Serializer\\Mapping\\AttributeMetadata'],
+        clone $p['Symfony\\Component\\Serializer\\Mapping\\AttributeMetadata'],
```

Comment:
IIUC, there will be one such exported structure per class with metadata, isn't it? Then there might be a significant performance issue here: requiring this file will instantiate all metadata objects for all classes. This might load classe metadata that are not used and thus might impact performance by not being lazy enough.

I see two strategies to fix that:
1. wrap exported structures in closures - that's the easiest
2. ensure we export pure arrays, and create the attribute objects at runtime. This might be a bit harder but has the best performance potential I think (can be compared to the previous one since 1. should be easy to implement even just for the sake of the comparison)

Reply:
I've updated the code to use what I think pure arrays are. Is it what do you mean for 2nd strategy?